### ISSUE TITLE
With record set to "prog", program mode activated on boot after flash

### ISSUE THREAD

**Issue by tremby** [cdisselkoen/Kaleidoscope-MacrosOnTheFly#5]
I have record set to the "prog" key. After flashing (which I have to initiate by holding prog of course) the keyboard starts up in macro programming mode. The first key I then press lights up white.

I'm not yet sure whether this is every time or just if I hold the key too long.

**Comment by cdisselkoen** [comment]
In general, if you have `prog` mapped to a key in your sketch, anytime you press `prog` for any reason, Kaleidoscope will follow your orders and handle the keypress.  It has no way of knowing whether you're pressing it because you intend to flash, or because you want to record.  In this case, Kaleidoscope correctly notices you've pressed `Key_MacroRec` and enters recording mode.

If you are flashing the keyboard, as soon as the red light starts chasing across the keyboard it is safe to release `prog`.  If you release before it's done, you shouldn't have this issue.

That said, it might be possible to create a rule like "ignore the `Key_MacroRec` press if it seems like the key was held during boot".  That might be a desirable feature.

**Comment by tremby** [comment]
Looks like your first paragraph is responding to a different ticket (#6).

I could have sworn I stopped holding prog long before the flashing finished and still had this issue. I'll give it another try soon.

**Comment by cdisselkoen** [comment]
No, that applies to this ticket as well.  In this case, Kaleidoscope sees that `prog` is pressed immediately when it boots, and it immediately enters recording mode.

I'd be interested to know if this behavior occurs even if `prog` is released before flashing ends.
